packages/core/src/agent/index.ts

```diff
@@ -1299,12 +1361,22 @@ export class Agent<
                     resourceId,
                     config: memoryConfig,
                     // The new user messages aren't in the list yet cause we add memory messages first to try to make sure ordering is correct (memory comes before new user messages)
-                    vectorMessageSearch: new MessageList().add(messages, `user`).getLatestUserContent() || '',
+                    vectorMessageSearch: userContent,
                   })
                   .then(r => r.messagesV2),
                 memory.getSystemMessage({ threadId: threadObject.id, resourceId, memoryConfig }),
               ])
-            : [[], null, null];
+            : [[], null];
+
+        if (memoryMessages.length > 0) {
+          convertedTools.semanticRecall = this.getSemanticRecallTool({
```

Comment:
I don't think we want to do it this way. This is going to make the agent always call this tool. What we want to do is just simulate that the tool was called when we stream to playground

Reply:
The agent shouldn't have access to the tool - we're just using a tool call as a quick/easy way to add this to playground, but we definitely don't want to augment how agent memory works to accomplish that

Reply:
Changing this to just be a traced function that we can then display.